src/surge-xt/gui/overlays/LuaEditors.cpp

```diff
@@ -1010,7 +1021,455 @@ struct WavetablePreviewComponent : juce::Component
+            generateS->setEnabled(true);
+            addAndMakeVisible(*generateS);
+        }
+    }
+
+    std::unique_ptr<juce::Label> newL(const std::string &s)
+    {
+        auto res = std::make_unique<juce::Label>(s, s);
+        res->setText(s, juce::dontSendNotification);
+        res->setFont(skin->fontManager->getLatoAtSize(9, juce::Font::bold));
+        res->setColour(juce::Label::textColourId, skin->getColor(Colors::MSEGEditor::Text));
+        return res;
+    }
+
+    int32_t controlModifierClicked(GUI::IComponentTagValue *c, const juce::ModifierKeys &mods,
+                                   bool isDoubleClickEvent) override
+    {
+        auto tag = (tags)(c->getTag());
+
+        switch (tag)
+        {
+        case tag_select_tab:
+        case tag_code_apply:
+        case tag_generate_wt:
+        {
+            juce::PopupMenu contextMenu;
+
+            auto msurl = editor->helpURLForSpecial("formula-editor");
+            auto hurl = editor->fullyResolvedHelpURL(msurl);
+
+            editor->addHelpHeaderTo("Formula Editor", hurl, contextMenu);
+
+            contextMenu.showMenuAsync(editor->popupMenuOptions(this, false),
+                                      Surge::GUI::makeEndHoverCallback(c));
+        }
+        break;
+        default:
+            break;
+        }
+        return 0;
+    }
+
+    void valueChanged(GUI::IComponentTagValue *c) override
+    {
+        auto tag = (tags)(c->getTag());
+
+        switch (tag)
+        {
+        case tag_select_tab:
+        {
+            int m = c->getValue();
+
+            if (m > 0.5)
+            {
+                overlay->showPreludeCode();
+            }
+            else
+            {
+                overlay->showModulatorCode();
+            }
+        }
+        break;
+        case tag_code_apply:
+        {
+            overlay->applyCode();
+        }
+        break;
+        case tag_generate_wt:
+        {
+            overlay->generateWavetable();
+        }
+        break;
+        default:
+            break;
+        }
+    }
+
+    void comboBoxChanged(juce::ComboBox *comboBoxThatHasChanged) override
+    {
+        overlay->osc->wavetable_formula_res_base = resolutionB->getSelectedId();
+        overlay->rerenderFromUIState();
+    }
+
+    void textEditorReturnKeyPressed(juce::TextEditor &) override
+    {
+        overlay->osc->wavetable_formula_nframes = std::atoi(framesT->getText().toRawUTF8());
+        overlay->rerenderFromUIState();
+    }
+
+    std::unique_ptr<juce::Label> codeL, resolutionL, framesL;
+    std::unique_ptr<Surge::Widgets::MultiSwitchSelfDraw> codeS, applyS, generateS;
+    std::unique_ptr<juce::ComboBox> resolutionB;
+    std::unique_ptr<juce::TextEditor> framesT;
+
+    void paint(juce::Graphics &g) override { g.fillAll(skin->getColor(Colors::MSEGEditor::Panel)); }
+
+    void onSkinChanged() override { rebuild(); }
+
+    JUCE_DECLARE_NON_COPYABLE_WITH_LEAK_DETECTOR(WavetableScriptControlArea);
+};
```

Comment:
I didn't read this all that carefully but I am assuming it is ok since it looks copied from other similar with appropriate changes. But we should add an item to your work list before we ship 1.4 to do an accessibility check here (which I can help with).

Reply:
I will add this to the WTSE issue.